### PR TITLE
perf(ecmul): use GLV with safe handling of edge cases in EVM ecmul

### PR DESCRIPTION
**Pull request by yelhousni**
# Description

Another tiny PR to use GLV with safe handling of edge cases in EVM `ecmul` precompile.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

Tests in `bn_test.go` pass.
 
# How has this been benchmarked?

ECMUL precompile cost goes from **781,437** to **673,659** SCS.

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I did not modify files generated from templates
- [x] `golangci-lint` does not output errors locally
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by yelhousni** [comment]
@ivokub just re-pinging you here

File: std/algebra/emulated/sw_emulated/point.go [hunk 1]
```diff
@@ -532,13 +540,157 @@ func (c *Curve[B, S]) scalarMulGLV(Q *AffinePoint[B], s *emulated.Element[S], op
...
+		R0 = c.Select(sBits[i], Rb, R0)
+		R1 = c.Select(sBits[i], R1, Rb)
+	}
+
+	// i = n-1
+	Rb = c.doubleAndAddSelect(sBits[n-1], R0, R1)
+	R0 = c.Select(sBits[n-1], Rb, R0)
+
+	// i = 0
+	R0 = c.Select(sBits[0], R0, c.Add(R0, c.Neg(p)))
```
Comment (ivokub) [inline std/algebra/emulated/sw_emulated/point.go:579]: Do I understand correctly that here is the only difference for safe and unsafe? I feel the optional `Safe` option would make sense as we would only be changing one function. Then we can have less duplication.

**Review by ivokub — APPROVED** [review]
PR looks good. I'll accept it, but maybe I can try implementing the safe scalarmul with option instead to see how it looks. I have actually also thought about adding an option which would allow to define for MSM which inputs may be unsafe (i.e. 0 scalar or 0 point) and then these options would complement each other quite nicely.

**Comment by ivokub** [comment]
@yelhousni - I implemented using an option and merged the implementations. See how it looks and feels. If seems excessive then I'm good with reverting. On one hand I do not want to overuse using options, but I still like having unified implementation.

Another thing - currently the default is to use unsafe and we use option to use safe version. But I guess it would make more sense to have the defaults reversed -- i.e. without options we use safe formulas and then can optionally toggle to unsafe formulas for performance. I think we may have incoming bug reports due to the defaults being the other way.
But I didn't want to change now as it would have required a lot of refactoring all over, so dunno.
